### PR TITLE
Better deserialization errors

### PR DESCRIPTION
**Pull request by cameron1024**
Replaces a few custom deserializers with manual `Deserialize` implementations (with descriptions of the format of the data for better error messages)

An example error:
```
Error:
   0: error decoding response body: didn't match `$N in ada` pattern at line 1 column 146
   1: didn't match `$N in ada` pattern at line 1 column 146

Location:
   catalyst-toolbox/src/bin/cli/ideascale/mod.rs:119
```

### CODE REVIEW

File: script.sh [hunk 1]
```diff
@@ -0,0 +1 @@
+cargo run --bin catalyst-toolbox ideascale import --api-token 3ff50be1-084c-47b1-9df5-53aa2e150099 --fund 8  --chain-vote-type private --output-dir . --threshold 450 --fund-goal "Create, fund and deliver the future of Cardano." --excluded-proposals ../catalyst-resources/ideascale/fund8/excluded_proposals.json --tags ../catalyst-resources/ideascale/fund8/tags.json  --stages-filters "Assess QA"
```
Comment (Zeegomo) [inline script.sh:1]: We probably don't need this?

File: catalyst-toolbox/src/ideascale/models/de/ada_rewards.rs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        // missing dollar sign
+        assert!(parse("123 in ada").is_err());
+        // negative number
+        assert!(parse("$-123 in ada").is_err());
+        // fraction
+        assert!(parse("$123.0 in ada").is_err());
+    }
+}
+
+// fn deserialize_rewards<'de, D: Deserializer<'de>>(deserializer: D) -> Result<u64, D::Error> {
```
Comment (Zeegomo) [inline catalyst-toolbox/src/ideascale/models/de/ada_rewards.rs:106]: I guess this is not needed anymore?

File: catalyst-toolbox/src/ideascale/models/de/approval.rs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+        formatter.write_str("a string representing an approval status (with ths string \"approved\" meaning `Approved`, and all other strings being `NotApproved`)")
+    }
+
+    fn visit_str<E>(self, v: &str) -> Result<Self::Value, E>
+    where
+        E: serde::de::Error,
+    {
+        match v {
+            "approved" => Ok(Approval::Approved),
+            _ => Ok(Approval::NotApproved),
```
Comment (Zeegomo) [inline catalyst-toolbox/src/ideascale/models/de/approval.rs:57]: nit: `"Approved"` would be parsed as `NotApproved` which is a bit counterintuitive, I think it's better to have this in lowercase before comparison

**Review by Zeegomo — CHANGES_REQUESTED** [review]

File: catalyst-toolbox/Cargo.toml [hunk 4]
```diff
@@ -61,9 +60,10 @@ symmetric-cipher = { git = "https://github.com/input-output-hk/chain-wallet-libs
 graphql_client = { version = "0.10" }
 gag = "1"
 vit-servicing-station-lib = { git = "https://github.com/input-output-hk/vit-servicing-station.git", branch = "master" }
-env_logger = "0.9"
 voting-hir = { path = "../voting-hir", features = ["serde"] }
 fraction = "0.10"
+tracing = "0.1"
+tracing-subscriber = "0.3"
```
[the remark this answers is not in the document]
Comment (cameron1024) [reply catalyst-toolbox/Cargo.toml:66]: I added it for debugging something, but I don't think it's strictly needed any more. Happy to remove, but I guess it probably doesn't make too much difference whether we use `tracing` or `log` :shrug:

File: script.sh (hunk 1 above)
Comment (cameron1024) [reply script.sh:1]: Oops

File: .pre-commit-config.yaml [hunk 5]
```diff
@@ -0,0 +1 @@
+/nix/store/80r7admw5vnzl9zxhbznd1yqlfmajck1-pre-commit-config.json
```
Comment (Zeegomo) [inline .pre-commit-config.yaml:1]: should this be committed?

**Comment by Mr-Leshiy** [comment]
need to update with the latest master :)

**Review by saibatizoku — APPROVED** [review]
tracing! awesome

**Review by Mr-Leshiy — APPROVED** [review]

**Pull request merged**